DS-808 Profile edit form mockup

MarcinMr opened the pull request:
## Jira

https://pegadigitalit.atlassian.net/browse/DS-808

## Summary

A mockup page was created in the User folder -> `/pattern-lab/?p=pages-user-profile-edit`

## Details

A Profile Edit Form was created similar to the design attached in the  [Jira task](https://pegadigitalit.atlassian.net/browse/SN-243). The card's design is omitted, instead, a simple page with form inputs is created.

## How to test

Pull the branch. Make sure that the form is created in a proper way and all necessary inputs are included.

mikemai2awesome reviewed (changes requested):
@MarcinMr Good job with this. Please fix the following issues:

1. Add a page heading with `h1`. It can be "Edit Profile".
1. Heading orders: each section should be an `h2`, subsequent headings in each section should be `h3`.

    <img width="340" alt="Screen Shot 2022-06-24 at 12 20 39 PM" src="https://user-images.githubusercontent.com/3027663/175576931-b7a21670-4598-41ab-958e-55dfd5a30601.png">

1. Fix missing labels for certain inputs.

    <img width="1533" alt="Screen Shot 2022-06-24 at 12 21 44 PM" src="https://user-images.githubusercontent.com/3027663/175577604-b78cc8de-f06f-4690-8cab-3dc6e3cf5cf4.png">

You can use either Sa11y bookmarklet or Chrome WAVE extension to detect these issues.

MarcinMr commented:
@mikemai2awesome 
- H1 is added and the hierarchy of the headings is fixed. 
- Fixed input labels
- Moved page from the sandbox to the User folder

mikemai2awesome reviewed (approved):
All set from me.

krlucas commented:
@MarcinMr @mikemai2awesome 
This looks great. A couple of requests/comments.

* Can we up with examples of two or more existing
** Projects
*** With multiple products already added
** Languages
** Project Roles
** Pega Specializations
* Should Project > Product, Roles and Specializations use the React multi-select? cc @danielamorse

ajfwebdev commented on docs-site/src/pages/pattern-lab/_patterns/50-pages/85-user/-user-profile-edit.twig:
```diff
@@ -0,0 +1,1535 @@
+            displayType: 'block',
+            attributes: {
+              for: 'phone-number'
+            }
```
@MarcinMr Based on the image of the mockup, we'll need some changes for phone number editing. It should work like it does today on the user profile edit form on MyPega. Pega users like us are not allowed to edit the phone number, so we can only see the current phone number string (including country dial code) and the extension, if any. If you login to QA as a non-pega user, you will see a button to click if you wish to edit the phone number, and new fields for editing will appear. There will be one select field for selecting the country dial code, one field for typing in the rest of the phone number, and a third for typing in the extension.

krlucas replied:
This is how the phone number works in MyPega
![2022-06-27_17-08-50](https://user-images.githubusercontent.com/150459/176036156-77466439-2ab8-434b-9364-e82eb205eaa1.png)

Then click the pencil Icon
![2022-06-27_17-09-18](https://user-images.githubusercontent.com/150459/176036244-889d7e5a-348a-4649-8b5c-8806063f46b2.png)

Search for a country
![2022-06-27_17-09-36](https://user-images.githubusercontent.com/150459/176036278-e7e34ece-af32-4ede-a7b1-7721cd9c8fa5.png)

MarcinMr replied:
@ajfwebdev @krlucas 

I've made some changes here related to the editing phone number fieldset. Is it what we are looking for?

krlucas commented:
@MarcinMr I'll defer to @ajfwebdev and Tomasz on the phone number. It looks workable to me.

I'd still like you update with an example of a user who has existing Projects, Languages, Roles and Pega Specializations.

Here's a screenshot of what that looks like in MyPega
<img width="565" alt="2022-06-28_09-57-27" src="https://user-images.githubusercontent.com/150459/176197413-afed17ce-4161-4625-abbe-0a7fab4ed92d.png">

MarcinMr commented:
@krlucas thanks for the screenshot. I updated the mockup with the existing user projects and selected languages. I cannot update the `Project Roles` and `Pega Specializations` like on the screen. In this case, the multiselect is the solution, but it only lives in Drupal. I added an information banner that says to use pega_react_multi_select module. The info is also added by Dan in the comment on [ticket SN-243](https://pegadigitalit.atlassian.net/browse/SN-243)

krlucas commented:
@MarcinMr Thanks. Those updates are helpful.

MyPega supports reordering existing projects via drag and drop.
<img width="543" alt="2022-06-28_09-57-27 copy" src="https://user-images.githubusercontent.com/150459/176694351-b41fbc21-e488-41a9-9b65-7674b50ee9c5.png">

I don't understand the placement of the drag handle in your mockup. It should go next to each existing project.
![2022-06-30_09-51-45](https://user-images.githubusercontent.com/150459/176694741-c40acd17-8dbf-4f08-9942-79ac21d38398.png)

cc @colbytcook

colbytcook commented:
@MarcinMr I am not sure of the mobile behavior of the "My Experience" section, particularly the vertical stacking of the "drag", "content" and the "edit" options. Should maintain a horizontal layout throughout?

colbytcook commented on docs-site/src/pages/pattern-lab/_patterns/50-pages/85-user/-user-profile-edit.twig:
```diff
@@ -0,0 +1,2095 @@
+    padding_top: 'large',
+    padding_bottom: 'large',
+    template: [
+      '75@from-small',
```
@MarcinMr after reviewing the responsive behavior I think we should switch this to '75@from-medium' in order to give the form a little bit more breathing room around ~700 pixels device view size.

colbytcook replied:
Updated, closing this comment
